### PR TITLE
[AR-2240] fetch app config and integrate with the wallet

### PR DESCRIPTION
**Pull request by karthik-durai**
1. Fetch app info like name, theme, and logo images
2. Apply them in respective places

### CODE REVIEW

File: src/network.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import axios from 'axios'
+import { ImagesURLs, Theme } from './interfaces'
+
+const BASE_URL = 'https://gateway-dev.arcana.network/api/'
+
+function getImageUrls(appId: string, theme: Theme): ImagesURLs {
```
Comment (ajithranka) [inline src/network.ts]: Inline the return type.

```suggestion
function getImageUrls(appId: string, theme: Theme): { horizontal: string, vertical: string } {
```

File: src/network.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import axios from 'axios'
+import { ImagesURLs, Theme } from './interfaces'
+
+const BASE_URL = 'https://gateway-dev.arcana.network/api/'
```
Comment (ajithranka) [inline src/network.ts]: Check if this needs to be configurable.

File: src/network.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+function getImageUrls(appId: string, theme: Theme): ImagesURLs {
+  const url = `https://gateway-dev.arcana.network/api/v2/app/${appId}/logo?type=${theme}`
```
Comment (ajithranka) [inline src/network.ts]:
```suggestion
  const url = `${BASE_URL}/v2/app/${appId}/logo?type=${theme}`
```

File: src/index.ts [hunk 4]
```diff
@@ -29,21 +30,49 @@ class WalletProvider {
...
-  public async init(themeConfig: IWidgetThemeConfig) {
+  public async init() {
     if (this.iframeWrapper) {
       return
     }
+
+    const appId = this.params.appId
+
+    const APP_INFO_API = 'v1/get-app-theme/'
+    const AppInfoApiParams = { id: appId }
```
Comment (ajithranka) [inline src/index.ts]: Inline at the point of use.

File: src/index.ts [hunk 5]
```diff
@@ -29,21 +30,49 @@ class WalletProvider {
...
+    const { data: appInfo }: { data: AppInfo } = await network.get(
+      APP_INFO_API,
+      {
+        params: AppInfoApiParams,
+      }
+    )
+
+    const appImageURLs: ImagesURLs = getImageUrls(appId, appInfo.theme)
+
+    const appConfig: IAppConfig = {
```
Comment (ajithranka) [inline src/index.ts:44]: Check if `IAppConfig` is exported from the wallet.

File: src/interfaces.ts [hunk 6]
```diff
@@ -12,19 +17,23 @@ export interface IframeWrapperParams {
...
-        vertical: string
-      }
+      horizontal: string
+      vertical: string
     }
   }
   theme: Theme
 }
 
+export type AppInfo = {
```
Comment (ajithranka) [inline src/interfaces.ts]: Let's stick to interfaces.

```suggestion
export interface IAppInfo = {
```

**Review by ajithranka — APPROVED** [review]

**Comment by makylfang** [comment]
@karthik-durai 
Any reason to use axios instead of fetch?

File: src/network.ts (hunk 2 above)
Comment (makylfang) [inline src/network.ts]: Use `getConfig().GATEWAY_URL` (getConfig from `config.ts`) instead of `BASE_URL`

**Comment by karthik-durai** [comment]
> @karthik-durai Any reason to use axios instead of fetch?

we are/will be using `axios` on front-end projects, to achieve consistency I used `axios`.

**Comment by makylfang** [comment]
> > @karthik-durai Any reason to use axios instead of fetch?
> 
> we are/will be using `axios` on front-end projects, to achieve consistency I used `axios`.

The SDK already uses fetch for one of the other calls, don't want to import a dependency for one call, do you mind changing it to fetch?

File: src/index.ts (hunk 5 above)
Comment (karthik-durai) [reply src/index.ts:44]: added to export obj in `index.ts`

**Comment by karthik-durai** [comment]
> > > @karthik-durai Any reason to use axios instead of fetch?
> > 
> > 
> > we are/will be using `axios` on front-end projects, to achieve consistency I used `axios`.
> 
> The SDK already uses fetch for one of the other calls, don't want to import a dependency for one call, do you mind changing it to fetch?

ok.

File: src/network.ts (hunk 2 above)
Comment (karthik-durai) [reply src/network.ts]: used `getConfig` function from `config.ts`.

**Review by makylfang — APPROVED** [review]
LGTM

**Pull request merged**